sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_client_manager.py

```diff
@@ -380,8 +422,16 @@ def _setup_failover_endpoints(self):
                             **self._args
                         )
                     )
-        self._replica_clients = new_clients
         self._next_update_time = time.time() + MINIMAL_CLIENT_REFRESH_INTERVAL
+        if updated_endpoints and not self._load_balancing_enabled:
+            random.shuffle(discovered_clients)
+            self._replica_clients = [self._original_client] + discovered_clients
+        if updated_endpoints and self._load_balancing_enabled:
+            # Reshuffle only if a new client was added and _load_balancing_enabled is enabled
+            # This allways needs to be shuffled, but if load balancing is enabled than the primary endpoint need to be
+            #  shuffled too.
+            self._next_update_time = time.time() + MINIMAL_CLIENT_REFRESH_INTERVAL
+            random.shuffle(self._replica_clients)
```

Comment:
```suggestion
        self._next_update_time = time.time() + MINIMAL_CLIENT_REFRESH_INTERVAL
        
        if not self._load_balancing_enabled:
            random.shuffle(discovered_clients)
            self._replica_clients = [self._original_client] + discovered_clients
        else:
            self._replica_clients = [self._original_client] + discovered_clients
            random.shuffle(self._replica_clients)
```

Reply:
Don't we want to only reshuffle if new clients are found?

Reply:
Why? Replica addition/deletion is not a frequent operation. So discovered replicas list may rarely change. But we still want to shuffle the replicas from time to time.

Reply:
@amerjusupovic what do we do in .Net? The time between checks for new clients is once an hour. So, every hour we re-shuffle?

Reply:
Yes, if it's been at least an hour since the last refresh and a refresh is triggered, then we refresh the clients.